examples/server_side_filtering.rs

```diff
@@ -0,0 +1,144 @@
+            let revision = event.get_original_event().revision;
+
+            println!("Received event {}@{}", revision, stream_id);
+        }
+    }
+    //endregion exclude-system
+
+    Ok(())
+}
+
+pub async fn event_type_prefix(client: &Client) -> Result<()> {
+    //region event-type-prefix
+    let filter = SubscriptionFilter::on_event_type().add_prefix("customer-");
+    let options = SubscribeToAllOptions::default().filter(filter);
+
+    let mut sub = client.subscribe_to_all(&options).await?;
+    //endregion event-type-prefix
+
+    while let Some(event) = sub.try_next().await? {
+        if let SubEvent::EventAppeared(event) = event {
+            let stream_id = event.get_original_stream_id();
+            let revision = event.get_original_event().revision;
+
+            println!("Received event {}@{}", revision, stream_id);
+        }
+    }
+
+    Ok(())
+}
+
+pub async fn event_type_regex(client: &Client) -> Result<()> {
+    //region event-type-regex
+    let filter = SubscriptionFilter::on_event_type().regex("^user|^company");
+    let options = SubscribeToAllOptions::default().filter(filter);
+
+    let mut sub = client.subscribe_to_all(&options).await?;
+    //endregion event-type-regex
+
+    while let Some(event) = sub.try_next().await? {
+        if let SubEvent::EventAppeared(event) = event {
+            let stream_id = event.get_original_stream_id();
+            let revision = event.get_original_event().revision;
+
+            println!("Received event {}@{}", revision, stream_id);
+        }
+    }
+
+    Ok(())
+}
+
+pub async fn stream_prefix(client: &Client) -> Result<()> {
+    //region stream-prefix
+    let filter = SubscriptionFilter::on_stream_name().add_prefix("user-");
+    let options = SubscribeToAllOptions::default().filter(filter);
+
+    let mut sub = client.subscribe_to_all(&options).await?;
+    //endregion stream-prefix
+
+    while let Some(event) = sub.try_next().await? {
+        if let SubEvent::EventAppeared(event) = event {
+            let stream_id = event.get_original_stream_id();
+            let revision = event.get_original_event().revision;
+
+            println!("Received event {}@{}", revision, stream_id);
+        }
+    }
+
+    Ok(())
+}
+
+pub async fn stream_regex(client: &Client) -> Result<()> {
+    //region stream-regex
+    let filter = SubscriptionFilter::on_event_type().regex("/^[^\\$].*/");
+    let options = SubscribeToAllOptions::default().filter(filter);
+
+    let mut sub = client.subscribe_to_all(&options).await?;
+    //endregion stream-regex
+
+    while let Some(event) = sub.try_next().await? {
+        if let SubEvent::EventAppeared(event) = event {
+            let stream_id = event.get_original_stream_id();
+            let revision = event.get_original_event().revision;
+
+            println!("Received event {}@{}", revision, stream_id);
+        }
+    }
+
+    Ok(())
+}
+
+pub async fn checkpoint_callback_with_interval(client: &Client) -> Result<()> {
```

Comment:
**FIX:** Missing checkpoint sample https://developers.eventstore.com/clients/grpc/subscribing-to-streams/filtering.html#checkpointing.

Reply:
It's available on line 135

Reply:
It's missing the `//region checkpoint` - see, e.g. https://github.com/EventStore/EventStoreDB-Client-Java/blob/5195449b17eda763adce48f45b8e7218ef4039d9/db-client-java/src/test/java/com/eventstore/dbclient/samples/server_side_filtering/ServerSideFiltering.java#L135.

Reply:
Using `//region checkpoint` around while region.

Reply:
done